### PR TITLE
Opitimization animation parser code and improve paser performance

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Review by luzhuang — APPROVED** [review]
+1

File: packages/core/src/animation/animationCurve/AnimationCurve.ts [hunk 1]
```diff
@@ -51,12 +51,18 @@ export abstract class AnimationCurve<V extends KeyframeValueType> {
...
-    this.keys.push(key);
-    if (time > this._length) {
+    const { keys } = this;
+
+    if (time >= this._length) {
+      keys.push(key);
       this._length = time;
+    } else {
+      let index = keys.length;
+      while (index > 0 && key.time < keys[index - 1].time) {
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/animation/animationCurve/AnimationCurve.ts:61]: while(--index > 0) {
   if( time < keys[index]) {
       keys.splice(index, 0, key);
       break;
   }
}

Comment (GuoLei1990) [reply packages/core/src/animation/animationCurve/AnimationCurve.ts:61]: image
This opt has bug

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Pull request merged**